### PR TITLE
LibWeb: Add basic MathML support

### PR DESCRIPTION
**Pull request by Epigenetic**
We now can parse and generate MathML elements using the [MathML Core](https://w3c.github.io/mathml-core/#the-mathvariant-attribute) standard. We don't have any of the MathML CSS extensions yet though, so it looks a bit wonky most of the time.

### CODE REVIEW

File: Userland/Libraries/LibWeb/Bindings/MainThreadVM.cpp [hunk 1]
```diff
@@ -95,6 +96,7 @@ ErrorOr<void> initialize_main_thread_vm()
     TRY(UIEvents::EventNames::initialize_strings());
     TRY(WebGL::EventNames::initialize_strings());
     TRY(XHR::EventNames::initialize_strings());
+    TRY(MathML::TagNames::initialize_strings());
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibWeb/Bindings/MainThreadVM.cpp:99]: This list is alphabetical. :redlinus:

File: Userland/Libraries/LibWeb/DOM/ElementFactory.cpp [hunk 2]
```diff
@@ -474,6 +476,44 @@ static WebIDL::ExceptionOr<JS::GCPtr<SVG::SVGElement>> create_svg_element(JS::Re
     return nullptr;
 }
 
+static WebIDL::ExceptionOr<JS::GCPtr<MathML::MathMLElement>> create_mathml_element(JS::Realm& realm, Document& document, QualifiedName qualified_name)
+{
+    auto const& local_name = FlyString::from_deprecated_fly_string(qualified_name.local_name()).release_value_but_fixme_should_propagate_errors();
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibWeb/DOM/ElementFactory.cpp:481]: You can propagate this error with `TRY_OR_THROW_OOM(realm.vm(), ...)`

File: Userland/Libraries/LibWeb/DOM/ElementFactory.cpp [hunk 3]
```diff
@@ -474,6 +476,44 @@ static WebIDL::ExceptionOr<JS::GCPtr<SVG::SVGElement>> create_svg_element(JS::Re
...
+
+    if (local_name == MathML::TagNames::annotation
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibWeb/DOM/ElementFactory.cpp:483]: Nit-pick: `local_name.is_one_of(...)` might be a little less verbose, but it's no big deal, so feel free to ignore this.

**Review by AtkinsSJ — CHANGES_REQUESTED** [review]
Just nit picks really.

**Comment by AtkinsSJ** [comment]
This needs some fixing up because of the recent error-propagation changes, but seems good otherwise.

**Comment by Epigenetic** [comment]
> This needs some fixing up because of the recent error-propagation changes, but seems good otherwise.

The error prorogation changes should be all cleaned up now.

**Comment by AtkinsSJ** [comment]
`Tests/LibWeb/Layout/input/css-namespace-tag-name-selector.html` also needs rebaselining, because your changes actually give the `<math>` element some style properties.

**Comment by AtkinsSJ** [comment]
This is getting ridiculous, but this now has conflicts because of a cmake change. 😆

**Comment by Epigenetic** [comment]
> This is getting ridiculous, but this now has conflicts because of a cmake change. 😆

Alright that's cleaned up now. Fingers crossed that nothing else breaks before the CI finishes running but with the luck this PR has had... well nothing would surprise me.

**Comment by AtkinsSJ** [comment]
Against all the odds, it's merged! 🎉

**Pull request merged**